tst/EnergyPlus/unit/api/datatransfer.unit.cc

```diff
@@ -0,0 +1,608 @@
+            SetupOutputVariable(val.varName, OutputProcessor::Unit::kg_s, val.value, "Zone", "Average", val.varKey);
+        }
+    }
+
+    enum class ActuatorType {REAL, INTEGER, BOOL};
+    void preRequestActuator(std::string const &objType,
+                            std::string const &controlType,
+                            std::string const &objKey,
+                            ActuatorType t)
+    {
+        switch (t) {
+        case ActuatorType::REAL:
+            this->realActuatorPlaceholders.emplace_back(objType, controlType, objKey);
+            break;
+        case ActuatorType::INTEGER:
+            this->intActuatorPlaceholders.emplace_back(objType, controlType, objKey);
+            break;
+        case ActuatorType::BOOL:
+            this->boolActuatorPlaceholders.emplace_back(objType, controlType, objKey);
+            break;
+        }
+    }
+
+    void setupActuatorsOnceAllAreRequested() {
+        for (auto & act : this->realActuatorPlaceholders) {
+            SetupEMSActuator(act.objType, act.key, act.controlType, "kg/s", act.flag, act.val);
+        }
+        for (auto & act : this->intActuatorPlaceholders) {
+            SetupEMSActuator(act.objType, act.key, act.controlType, "kg/s", act.flag, act.val);
+        }
+        for (auto & act : this->boolActuatorPlaceholders) {
+            SetupEMSActuator(act.objType, act.key, act.controlType, "kg/s", act.flag, act.val);
+        }
+    }
+
+    void addInternalVariable(std::string const &varType, std::string const &varKey)
+    {
+        this->internalVarPlaceholders.emplace_back();
+        SetupEMSInternalVariable(varType, varKey, "kg/s", this->internalVarPlaceholders.back());
+        this->internalVarPlaceholders.clear();
+    }
+
+    void addPluginGlobal(std::string const &varName)
+    {
+        this->pluginManager.addGlobalVariable(varName);
+    }
+
+    void addTrendWithNewGlobal(std::string const &newGlobalVarName, std::string const &trendName, int numTrendValues)
+    {
+        this->pluginManager.addGlobalVariable(newGlobalVarName);
+        int i = EnergyPlus::PluginManagement::PluginManager::getGlobalVariableHandle(newGlobalVarName, true);
+        EnergyPlus::PluginManagement::trends.emplace_back(trendName, numTrendValues, i);
+    }
+
+    static void simulateTimeStepAndReport()
+    {
+        UpdateMeterReporting(OutputFiles::getSingleton());
+        UpdateDataandReport(OutputProcessor::TimeStepType::TimeStepZone);
+    }
+};
+
+TEST_F(DataExchangeAPIUnitTestFixture, DataTransfer_TestListAllDataInCSV)
+{
+
+    std::string const idf_objects = delimited_string({"Version, 9.3.0;"});
+    ASSERT_TRUE(process_idf(idf_objects, false)); // this had to be here or I was getting a strange segfault during a JSON string dtor
+
+    // first off, the function should return, even if there isn't anything meaningful in it (it will have headers)
+    std::string csvDataEmpty = listAllAPIDataCSV();
```

Comment:
@Myoldmopar This is a memory leak. You implicitly convert to `std::string` but it does not take ownership of the returned `const char *`. You no longer have a pointer to that allocated memory and cannot `delete` it. That said, it isn't probably a huge deal for these unit tests but shows how easy it is to leak memory.